src/Microsoft.AspNet.Mvc.Core/ActionResults/ContentResult.cs

```diff
@@ -46,7 +60,7 @@ public class ContentResult : ActionResult
 
             if (Content != null)
             {
-                await response.WriteAsync(Content, contentTypeHeader.Encoding);
+                await response.WriteAsync(Content, new ResponseEncodingWrapper(encoding));
```

Comment:
:+1: ...when used with StreamWriter the preamble is being generated...will change it...